### PR TITLE
Waas best practices

### PR DESCRIPTION
**Pull request by moskalyk**

### CODE REVIEW

File: docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+Due to the fact that transactions are not instant and take time, it is recommended to include some type of loading UI change to prevent making additional transactions by the user. 
+
+Making multiple transactions by accident can also be prevented by implementing the use of a local mutex to hold state until a transaction is complete, only then making UI changes.
+
+## 4. Updating Allowed Origins to Prevent Misuse
+
+If in a web environment, always ensure your allowed origins match your intended production deployment. 
+
+When complete with testing, make sure to remove any localhost origins so others do not also do gain local access to your Embbedded Wallet environment.
```
Comment (ScreamingHawk) [inline docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx]: typo Embedded -> Embedded in a few places.

File: docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+## 5. Keeping Your WaasConfigKey Hidden for Unity or Unreal Engine
+
+Ensure not to publish your WaaS Config Key anywhere online (e.g. github), so that your Embedded Wallet instance is not used by someone not yourself, as allowed origins do not apply to mobile games because the applications contain hidden code from public users.
```
Comment (ScreamingHawk) [inline docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx:48]: If `WaasConfigKey` needs to be secret, should it also remain out of distributed client applications? e.g. exes and webpages. Only used by the application server.

File: docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+## 6. Memetic Balance & Data Ownership
+
+In the transition from Universal Wallets to Embedded Wallets, an Embbedded based system puts engagement first to the benefit of every user. With integrations, there are doors that lead to thresholds informationally, therefore setting boundaries is a key tool to use in communication and function. 
+
+### Why? 
+When accessing alchemical thermodynamic energies for various games, ensuring to construct complimentary and adjusting practices of information or affordance can help maintain healthy ownership. As an example, conserving features that include: safety, accessibility, and intended value via a conserved structure in memes can better accomodate social settings.
```
Comment (ScreamingHawk) [inline docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx]: `alchemical thermodynamic energies` too abstract?

File: docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+Developers implementing an Embedded Wallet experience for their users have the opportunity to build ecosystems based on their wallet-as-a-service instance. This means, if a user logs into one game, that same user with same login flow can log into another game and be provided the same wallet address to sign or make transactions from.
+
+This is achieved by utilizing the same Client IDs (provided by Google or Apple) in these different gaming experiences registered in the [Sequence Builder](https://sequence.build) in order to yield the same authentication access flow for signing in users.
+
+## 3. Prevent Multiple Transactions by Repeat User Clicks
+
+Due to the fact that transactions are not instant and take time, it is recommended to include some type of loading UI change to prevent making additional transactions by the user. 
+
+Making multiple transactions by accident can also be prevented by implementing the use of a local mutex to hold state until a transaction is complete, only then making UI changes.
```
Comment (ScreamingHawk) [inline docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx:38]: Also consider batching transactions when multiple transactions are expected.

File: docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx (hunk 3 above)
Comment (JamesLawton) [reply docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx]: Typo in 'embbedded' wallet.

I don't understand this section. What is the goal here and what are we trying to communicate?

File: docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+## 6. Memetic Balance & Data Ownership
+
+In the transition from Universal Wallets to Embedded Wallets, an Embbedded based system puts engagement first to the benefit of every user. With integrations, there are doors that lead to thresholds informationally, therefore setting boundaries is a key tool to use in communication and function. 
+
+### Why? 
+When accessing alchemical thermodynamic energies for various games, ensuring to construct complimentary and adjusting practices of information or affordance can help maintain healthy ownership. As an example, conserving features that include: safety, accessibility, and intended value via a conserved structure in memes can better accomodate social settings.
+
+### Possible Solution(s)
+
+#### Internal Testing: Walk the Talk
```
Comment (JamesLawton) [inline docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx]: This also needs further clarification as I'm not sure what we are trying to convey.

File: docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+2. [Building an Ecosystem](/solutions/wallets/embedded-wallet/best-practices#2-building-an-ecosystem)
+3. [Prevent Multiple Transactions by Repeat User Clicks](/solutions/wallets/embedded-wallet/best-practices#3-prevent-multiple-transactions-by-repeat-user-clicks)
+4. [Updating Allowed Origins to Prevent Misuse](/solutions/wallets/embedded-wallet/best-practices#4-updating-allowed-origins-to-prevent-misuse)
+5. [Keeping Your WaasConfigKey Hidden for Unity or Unreal Engine](/solutions/wallets/embedded-wallet/best-practices#5-keeping-your-waasconfigkey-hidden-for-unity-or-unreal-engine)
+6. [Memetic Balance & Data Ownership](/solutions/wallets/embedded-wallet/best-practices#6-memetic-balance--data-ownership)
+
+## 1. Designs for Implementing a Collectible Mint
+
+### Using a Standard Builder Collectible Contract
+With developers implementing an Embedded Wallet experience for their users that have the goal to create a minting function using a contract, developers have the options to either: (1) offer sponsored cloudflare workers minting as a feature, or, (2) have an allowed list of wallet addresses.
```
Comment (JamesLawton) [inline docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx:17]: Developers could also make a paid mint - I don't think these are the only options. I also suggest removing 'cloudflare' as there are other alternatives to doing a serverless mint (even though we have a guide covering that one specifically)

**Review by JamesLawton — CHANGES_REQUESTED** [review]

File: docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx (hunk 4 above)
Comment (moskalyk) [reply docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx:38]: you can't batch transact with embedded wallet, as far as im aware. if so, we need to update our docs

File: docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx (hunk 2 above)
Comment (moskalyk) [reply docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx:48]: What do you mean by application server? waas config lives in the frontend

File: docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx (hunk 3 above)
Comment (moskalyk) [reply docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx]: it feels like embedded wallets get you closer to the technology, but there are side effects that might need adjusting through things like marketing or additional interaction that a game studio should think about

i updated this to 
> When accessing possibilities of experience for various games

File: docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx (hunk 5 above)
Comment (moskalyk) [reply docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx]: updated to 

> #### Internal Testing: Practicing your values

File: docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx (hunk 2 above)
Comment (ScreamingHawk) [reply docs/pages/solutions/wallets/embedded-wallet/best-practices.mdx:48]: If it's client side it's insecure and you should assume it's public knowledge. No reason to hide config from github. 
If it shouldn't be public, it shouldn't be client side.

**Pull request closed without merging**